### PR TITLE
Fix parseSoundPathString unit test

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR fixes a unit test for the `parseSoundPathString` function and follows up [this investigation PR](https://github.com/code-dot-org/code-dot-org/pull/51760). The `parseSoundPathString` function was added by [this PR](https://github.com/code-dot-org/code-dot-org/pull/51605). 

The first unit test in the `p5lab/utilsTest.js` file was failing inconsistently on the test machine. See [Slack thread](https://codedotorg.slack.com/archives/C03CM903Y/p1683562325265269).
When I ssh'd on to the test machine, the unit test failed every time I ran it using `npm run test:entry...` I saw that the `replaceAll` function was not recognized, so I used `replace` with a global variable instead. This fixed the issue on the test machine and the test passed consistently when run individually.

This presents to us a mystery - why does the unit test sometimes pass on the test machine? The pattern seems to be that when the test build [initially fails](https://codedotorg.slack.com/archives/C03CM903Y/p1683566699703729), if the test build is restarted, the unit test [does not fail](https://codedotorg.slack.com/archives/C03CM903Y/p1683572603364179). This makes me wonder if the unit tests are actually all run again on a rebuild.

I do want to note that I ran this unit test locally several times using `replaceAll` and the test passed each time. The MDN documentation states that [`replace`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#browser_compatibility) is supported for browser compatibility more broadly than [`replaceAll`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll#browser_compatibility). I wonder if this is impacting unit tests on the test machine?

To summarize, this PR fixes the unit test so it consistently passes locally AND on the test machine. However, I am unsure why the unit test using `replaceAll` sometimes passes during test builds even though the unit test consistently fails when run individually.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by breville — APPROVED** [review]

File: apps/src/p5lab/utils.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ export function parseSoundPathString(text) {
     category = pathStringArray[2];
     // Example: 'category_board_games' becomes 'Board games: '
     category = category.replace('category_', '');
-    category = category.replaceAll('_', ' ');
+    category = category.replace(/_/g, ' ');
```
Comment (mikeharv) [inline apps/src/p5lab/utils.js:41]: Clever idea to use `replace` with a regular expression instead. It seems like this would still replace all underscore characters in the string, is that right? For example, if we had `'category_fruits_and_vegetables'`, would the result be as expected? No category name in the current library is made of more than two English words, but I could imagine that changing at some point.

**Review by mikeharv — APPROVED** [review]
The best I can figure out is that the `replaceAll()` method was introduced in ECMAScript 2021, so if the JavaScript environment being used for the unit tests is using an older version of JavaScript is being used, then the original code could throw an error. I wonder if we should check the version of JS being used on the test machine browser.

File: apps/src/p5lab/utils.js (hunk 1 above)
Comment (fisher-alice) [reply apps/src/p5lab/utils.js:41]: Yes, this replaces all underscores with white spaces. For `category_fruits_and_vegetables` the result would be `Fruits and vegetables`. This is what I would think is expected. You?

**Comment by breville** [comment]
We might be able to resume using `replaceAll` since Node was coincidentally upgraded on our environments today!

**Comment by fisher-alice** [comment]
> 

Is there a benefit for going back to `replaceAll`? Is it readability?
I can also check on test if the unit test works with `replaceAll` some time today. Thanks!

**Comment by fisher-alice** [comment]
> > 
> 
> Is there a benefit for going back to `replaceAll`? Is it readability? I can also check on test if the unit test works with `replaceAll` some time today. Thanks!

Update: I just check on the test machine and `replaceAll` still is not recognized.
Screen Shot 2023-05-09 at 10 42 18 AM

**Comment by breville** [comment]
From the thread, it sounds like it's actually our environments' versions of `google-chrome-stable` that are little inconsistent, and so we shouldn't consider using `replaceAll` until they are updated.
